core/src/main/java/org/elasticsearch/index/query/QueryParsingException.java

```diff
@@ -44,7 +44,8 @@ public QueryParsingException(QueryParseContext parseContext, String msg, Object.
     }
 
     public QueryParsingException(QueryParseContext parseContext, String msg, Throwable cause, Object... args) {
-        super(parseContext.index(), msg, cause, args);
+        super(msg, cause, args);
```

Comment:
this is confusing and tricky with the `Object...` ctors they collide that's why we have this.. sorry something has to give